### PR TITLE
TELFE-221 View user profile

### PR DESCRIPTION
**Pull request by dptelicent**
Ticket: https://telicent.atlassian.net/browse/TELFE-221

### Goal

Amend the DS to allow the following (Graph PR to follow):

Given a user is viewing the Graph app

and /whoami is configured in the deployment

When a user looks at the top right of the screen

Then the user will see their user details in their user profile component as shown in the design.

and then app switcher will now be on the top left of the screen

and still work

if /whoami is not configured

Then the user will not see their user profile component on the top right of the screen.

### Work Completed

- User profile amended.
- App switcher is now in the top left when there is a user profile (previous location retained otherwise, in order not to potentially break functionality in other apps).
- Theme handling fixed.

### Testing Method

- Existing tests
- Testing added for new component.

### Engineering Notes 

This needs to be merged before the related Graph PR will pass pipeline tests (passing locally).

---

<details>
  <summary><strong>🕵️ Review Guidance</strong></summary>

---

General guidance

- Generally, approve a PR if it makes the system better, even if it's not perfect. — [Google: The Standard of Code Review](https://google.github.io/eng-practices/review/reviewer/standard.html)
- The aim of both PR AUTHOR and PR REVIEWER is to get the code merged
- Aim for consensus, defined as _everyone can live with the outcome_

For PR REVIEWER:

1. Read the ticket & description
2. [Review the code](https://google.github.io/eng-practices/review/reviewer/looking-for.html)
3. Request any changes that are essential.
4. For non-essential comments:
   - Use prefixes, e.g. "**nit:** change to Pascal case"
     - **nit:** small, non-essential change
     - **obs:** just an observation, doesn't affect the PR
     - **idea:** a suggestion to think about
     - **q:** questions
   - Use modifiers, e.g. "**obs**`[pr-owner-resolve]`: Jim is also editing this file"
     - `pr-author-resolve` PR author can resolve after reading
     - `pr-author-delete` (rare) delete after reading to avoid confusion
5. try to add _at least_ a helpful comment per ~500 lines; less if the PR is already busy

For PR AUTHOR:

1. Aim for enough detail in PR description for things to go smoothly
2. After requested changes, [re-request a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#re-requesting-a-review) (so the PR shows up in [reviews-requested:@me](https://github.com/pulls?q=is%3Apr+is%3Aopen+archived%3Afalse+sort%3Aupdated-desc+review-requested%3A%40me+))

</details>

### CODE REVIEW

File: src/v1/components/layout/AppChrome.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const AppChrome: React.FC<AppChromeProps> = ({ children, ...appBarProps }) => {
         maxWidth={false}
         disableGutters
         sx={{
-          height: "calc(100vh - 70px)",
+          height: `calc(100vh - ${theme.components?.MuiAppBar?.styleOverrides?.root?.height}px)`,
```
Comment (AlecsMissangu) [inline src/v1/components/layout/AppChrome.tsx:19]: What was the reason of this not being defined as 70px again?

**Comment by ACoolmanTelicent** [comment]
@dptelicent Could you do a screen recording showing the Ui changes? taa

File: src/v1/components/data-display/UserProfile/UserProfile.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import MUIMenu from "@mui/material/Menu";
 import { H2 } from "../Text/Text";
 import UserIcon from "../Icons/UserIcon";
 import DownArrowIcon from "../FontAwesomeIcons/DownArrowIcon";
+import "./user-profile.css";
```
Comment (AlecsMissangu) [inline src/v1/components/data-display/UserProfile/UserProfile.tsx:10]: Ash is keen on not using css in any of the v1 components. You will need to convert this to js styling. Also just checking if Justin is aware of the changes made here.

File: src/v1/theme/theme-type.d.ts
Comment (AlecsMissangu) [inline src/v1/theme/theme-type.d.ts:1]: Just out of curiosity, what is this adding? I'm not sure I understand from just looking at the code

**Review by AlecsMissangu — CHANGES_REQUESTED** [review]
Left some comments to improve my understanding. 
The css styles would need to changed to js styling.

File: src/v1/theme/theme-type.d.ts
Comment (dptelicent) [reply src/v1/theme/theme-type.d.ts:1]: Fixes theme handling so theme properties can be used where imported.

File: src/v1/components/data-display/UserProfile/UserProfile.tsx (hunk 2 above)
Comment (dptelicent) [reply src/v1/components/data-display/UserProfile/UserProfile.tsx]: JS styling now used.

Changes shown to @Justinki50.

File: src/v1/components/data-display/UserProfile/UserProfile.tsx [hunk 3]
```diff
@@ -25,11 +25,22 @@ const UserProfile: React.FC<UserProfileProps> = ({ fullName, children }) => {
...
+      max-width: 18rem;
+    }
+  `;
+
   return (
     <MUIStack direction="row" spacing={1} alignItems="center">
       <H2
         variant="h6"
-        className="username"
+        css={usernameStyle}
```
Comment (ACoolmanTelicent) [inline src/v1/components/data-display/UserProfile/UserProfile.tsx:43]: Hm. Not sure about css-to-js. Its perhaps ok - but I'm unsure.

For now lets stick to these "MUI blessed" methods of styling (in order of consideration):

1. [One-off customization](https://mui.com/material-ui/customization/how-to-customize/#1-one-off-customization)
1. [Reusable component](https://mui.com/material-ui/customization/how-to-customize/#2-reusable-component)
1. [Global theme overrides](https://mui.com/material-ui/customization/how-to-customize/#3-global-theme-overrides)
1. [Global CSS override](https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override)

So perhaps based on [response values](https://mui.com/system/getting-started/usage/#responsive-values)

```txx
<H2
sx={{
  display: { xs: 'none', sm: 'block' },
  maxWidth: {
    sm: '11rem',  // Applies when the screen width is 600px or more
    md: '18rem'   // Applies when the screen width is 960px or more
  }
}}
```

**Review by ACoolmanTelicent — CHANGES_REQUESTED** [review]

File: src/v1/components/data-display/UserProfile/UserProfile.tsx (hunk 3 above)
Comment (dptelicent) [reply src/v1/components/data-display/UserProfile/UserProfile.tsx]: Done.

**Review by ACoolmanTelicent — APPROVED** [review]

**Review by ACoolmanTelicent — APPROVED** [review]
lgtm

**Comment by ACoolmanTelicent** [comment]
I'm quite certain @AlecsMissangu would be cool with the changes. Hit me up if you have lingering concerns.

**Pull request merged**